// @atlaskit/modal-dialog/dist/cjs/components/Footer.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _button = _interopRequireDefault(require("@atlaskit/button"));

var _Content = require("../styled/Content");

var JustifyShim = function JustifyShim(props) {
  return _react.default.createElement("span", props);
};

var ModalFooter =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(ModalFooter, _Component);

  function ModalFooter() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, ModalFooter);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(ModalFooter)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "props", void 0);
    return _this;
  }

  (0, _createClass2.default)(ModalFooter, [{
    key: "render",
    // eslint-disable-line react/sort-comp
    value: function render() {
      var _this$props = this.props,
          actions = _this$props.actions,
          appearance = _this$props.appearance,
          component = _this$props.component,
          onClose = _this$props.onClose,
          showKeyline = _this$props.showKeyline;
      var warning = 'You can provide `component` OR `actions`, not both.';
      if (!component && !actions) return null;
      if (component && actions) return console.warn(warning); // eslint-disable-line no-console

      if (component) {
        return (0, _react.createElement)(component, {
          appearance: appearance,
          onClose: onClose,
          showKeyline: showKeyline
        });
      }

      return _react.default.createElement(_Content.Footer, {
        showKeyline: showKeyline
      }, _react.default.createElement(JustifyShim, null), _react.default.createElement(_Content.Actions, null, actions ? actions.map(function (_ref, idx) {
        var text = _ref.text,
            rest = (0, _objectWithoutProperties2.default)(_ref, ["text"]);
        var variant = idx ? 'subtle' : appearance || 'primary';
        return _react.default.createElement(_Content.ActionItem, {
          key: text || idx
        }, _react.default.createElement(_button.default, (0, _extends2.default)({
          appearance: variant
        }, rest), text));
      }) : null));
    }
  }]);
  return ModalFooter;
}(_react.Component);

exports.default = ModalFooter;
