// @atlaskit/modal-dialog/dist/cjs/components/Modal.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ModalDialogWithoutAnalytics = exports.Positioner = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _exenv = require("exenv");

var _analyticsNext = require("@atlaskit/analytics-next");

var _blanket = _interopRequireDefault(require("@atlaskit/blanket"));

var _package = require("../../package.json");

var _sharedVariables = require("../shared-variables");

var _Modal = require("../styled/Modal");

var _Animation = require("./Animation");

var _Content = _interopRequireDefault(require("./Content"));

var _FocusLock = _interopRequireDefault(require("./FocusLock"));

var Positioner = function Positioner(_ref) {
  var scrollBehavior = _ref.scrollBehavior,
      props = (0, _objectWithoutProperties2.default)(_ref, ["scrollBehavior"]);
  var PositionComponent = scrollBehavior === 'inside' ? _Modal.PositionerAbsolute : _Modal.PositionerRelative;
  return _react.default.createElement(PositionComponent, props);
};

exports.Positioner = Positioner;

function getScrollDistance() {
  return window.pageYOffset || document.documentElement && document.documentElement.scrollTop || document.body && document.body.scrollTop || 0;
}

var Modal =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(Modal, _Component);

  function Modal() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, Modal);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(Modal)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "state", {
      dialogNode: null,
      scrollDistance: _exenv.canUseDOM ? getScrollDistance() : 0,
      isExiting: false
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "handleWindowScroll", function () {
      if (getScrollDistance() !== _this.state.scrollDistance) {
        window.scrollTo(window.pageXOffset, _this.state.scrollDistance);
      }
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "handleOverlayClick", function (e) {
      if (_this.props.shouldCloseOnOverlayClick) {
        _this.props.onClose(e);
      }
    });
    return _this;
  }

  (0, _createClass2.default)(Modal, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var scrollDistance = getScrollDistance();

      if (getScrollDistance() !== this.state.scrollDistance) {
        // eslint-disable-next-line react/no-did-mount-set-state
        this.setState({
          scrollDistance: scrollDistance
        });
      }

      window.addEventListener('scroll', this.handleWindowScroll);
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      window.removeEventListener('scroll', this.handleWindowScroll);
    }
    /* Prevent window from being scrolled programatically so that the modal is positioned correctly
     * and to prevent scrollIntoView from scrolling the window.
     */

  }, {
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props = this.props,
          actions = _this$props.actions,
          appearance = _this$props.appearance,
          autoFocus = _this$props.autoFocus,
          body = _this$props.body,
          children = _this$props.children,
          footer = _this$props.footer,
          header = _this$props.header,
          height = _this$props.height,
          isChromeless = _this$props.isChromeless,
          isHeadingMultiline = _this$props.isHeadingMultiline,
          isOpen = _this$props.isOpen,
          onClose = _this$props.onClose,
          onCloseComplete = _this$props.onCloseComplete,
          onOpenComplete = _this$props.onOpenComplete,
          onStackChange = _this$props.onStackChange,
          shouldCloseOnEscapePress = _this$props.shouldCloseOnEscapePress,
          stackIndex = _this$props.stackIndex,
          heading = _this$props.heading,
          width = _this$props.width,
          scrollBehavior = _this$props.scrollBehavior;
      var scrollDistance = this.state.scrollDistance;
      var isBackground = stackIndex != null && stackIndex > 0; // If a custom width (number or percentage) is supplied, set inline style
      // otherwise allow styled component to consume as named prop

      var widthName = _sharedVariables.WIDTH_ENUM.values.includes(width) ? width : null;
      var widthValue = widthName ? null : width;
      return _react.default.createElement(_Animation.Animation, {
        in: isOpen,
        onExited: onCloseComplete,
        onEntered: onOpenComplete,
        stackIndex: stackIndex
      }, function (_ref2) {
        var fade = _ref2.fade,
            slide = _ref2.slide;
        return _react.default.createElement(_Modal.FillScreen, {
          style: fade,
          "aria-hidden": isBackground,
          scrollDistance: scrollDistance
        }, _react.default.createElement(_FocusLock.default, {
          isEnabled: stackIndex === 0 && isOpen,
          autoFocus: autoFocus
        }, _react.default.createElement(_blanket.default, {
          isTinted: true,
          onBlanketClicked: _this2.handleOverlayClick
        }), _react.default.createElement(Positioner, {
          style: slide,
          scrollBehavior: scrollBehavior,
          widthName: widthName,
          widthValue: widthValue
        }, _react.default.createElement(_Modal.Dialog, {
          heightValue: height,
          isChromeless: isChromeless,
          role: "dialog",
          tabIndex: "-1"
        }, _react.default.createElement(_Content.default, {
          actions: actions,
          appearance: appearance,
          footer: footer,
          heading: heading,
          isHeadingMultiline: isHeadingMultiline,
          header: header,
          onClose: onClose,
          shouldScroll: scrollBehavior === 'inside',
          shouldCloseOnEscapePress: shouldCloseOnEscapePress,
          onStackChange: onStackChange,
          isChromeless: isChromeless,
          stackIndex: stackIndex,
          body: body
        }, children)))));
      });
    }
  }]);
  return Modal;
}(_react.Component);

(0, _defineProperty2.default)(Modal, "defaultProps", {
  autoFocus: true,
  scrollBehavior: 'inside',
  shouldCloseOnEscapePress: true,
  shouldCloseOnOverlayClick: true,
  isChromeless: false,
  isOpen: true,
  stackIndex: 0,
  width: 'medium',
  isHeadingMultiline: true
});
var createAndFireEventOnAtlaskit = (0, _analyticsNext.createAndFireEvent)('atlaskit');
var ModalDialogWithoutAnalytics = Modal;
exports.ModalDialogWithoutAnalytics = ModalDialogWithoutAnalytics;

var _default = (0, _analyticsNext.withAnalyticsContext)({
  componentName: 'modalDialog',
  packageName: _package.name,
  packageVersion: _package.version
})((0, _analyticsNext.withAnalyticsEvents)({
  onClose: createAndFireEventOnAtlaskit({
    action: 'closed',
    actionSubject: 'modalDialog',
    attributes: {
      componentName: 'modalDialog',
      packageName: _package.name,
      packageVersion: _package.version
    }
  })
})(Modal));

exports.default = _default;
