// @atlaskit/modal-dialog/dist/cjs/components/ModalWrapper.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _theme = require("@atlaskit/theme");

var _portal = _interopRequireDefault(require("@atlaskit/portal"));

var _ModalTransition = require("./ModalTransition");

var _StackConsumer = _interopRequireDefault(require("./StackConsumer"));

var _Modal = _interopRequireDefault(require("./Modal"));

var ModalWrapper =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(ModalWrapper, _Component);

  function ModalWrapper() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, ModalWrapper);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(ModalWrapper)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "onModalClosed", function (onExited) {
      return function (e) {
        if (onExited) {
          onExited();
        }

        if (_this.props.onCloseComplete) {
          _this.props.onCloseComplete(e);
        }
      };
    });
    return _this;
  }

  (0, _createClass2.default)(ModalWrapper, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      return _react.default.createElement(_ModalTransition.ModalTransitionConsumer, null, function (_ref) {
        var isOpen = _ref.isOpen,
            onExited = _ref.onExited;
        return _react.default.createElement(_portal.default, {
          zIndex: _theme.layers.modal()
        }, _react.default.createElement(_StackConsumer.default, {
          isOpen: isOpen
        }, function (naturalStackIndex) {
          return _react.default.createElement(_Modal.default, (0, _extends2.default)({}, _this2.props, {
            isOpen: isOpen,
            stackIndex: _this2.props.stackIndex || naturalStackIndex,
            onCloseComplete: _this2.onModalClosed(onExited)
          }));
        }));
      });
    }
  }]);
  return ModalWrapper;
}(_react.Component);

(0, _defineProperty2.default)(ModalWrapper, "defaultProps", {
  autoFocus: true,
  scrollBehavior: 'inside',
  shouldCloseOnEscapePress: true,
  shouldCloseOnOverlayClick: true,
  isChromeless: false,
  width: 'medium',
  isHeadingMultiline: true,
  onClose: function onClose() {}
});
var _default = ModalWrapper;
exports.default = _default;
