// @atlaskit/modal-dialog/dist/cjs/components/ModalTransition.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.ModalTransitionConsumer = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _createContext = (0, _react.createContext)({
  isOpen: true,
  onExited: undefined
}),
    Consumer = _createContext.Consumer,
    Provider = _createContext.Provider; // checks if children exist and are truthy


var hasChildren = function hasChildren(children) {
  return _react.default.Children.count(children) > 0 && _react.default.Children.map(children, function (child) {
    return !!child;
  }).filter(Boolean).length > 0;
};

var Transition =
/*#__PURE__*/
function (_React$Component) {
  (0, _inherits2.default)(Transition, _React$Component);

  function Transition() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, Transition);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(Transition)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "state", {
      currentChildren: undefined
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "onExited", function () {
      _this.setState({
        currentChildren: _this.props.children
      });
    });
    return _this;
  }

  (0, _createClass2.default)(Transition, [{
    key: "render",
    value: function render() {
      return _react.default.createElement(Provider, {
        value: {
          onExited: this.onExited,
          isOpen: hasChildren(this.props.children)
        }
      }, this.state.currentChildren);
    }
  }], [{
    key: "getDerivedStateFromProps",
    value: function getDerivedStateFromProps(props, state) {
      var previousChildren = state.currentChildren;
      var exiting = hasChildren(previousChildren) && !hasChildren(props.children);
      return {
        currentChildren: exiting ? previousChildren : props.children
      };
    }
  }]);
  return Transition;
}(_react.default.Component);

var ModalTransitionConsumer = Consumer;
exports.ModalTransitionConsumer = ModalTransitionConsumer;
var _default = Transition;
exports.default = _default;
