// @atlaskit/modal-dialog/dist/cjs/styled/Content.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ActionItem = exports.Actions = exports.Footer = exports.Body = exports.TitleIconWrapper = exports.TitleText = exports.Title = exports.Header = exports.Wrapper = exports.keylineHeight = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _theme = require("@atlaskit/theme");

var _flexMaxHeightIeFix = require("../utils/flex-max-height-ie-fix");

// Constants
// ==============================
var innerGutter = 16;
var outerGutter = 20;
var keylineColor = (0, _theme.themed)({
  light: _theme.colors.N30,
  dark: _theme.colors.DN30
});
var keylineHeight = 2; // Wrapper
// ==============================

exports.keylineHeight = keylineHeight;

var Wrapper = _styledComponents.default.div.withConfig({
  displayName: "Content__Wrapper",
  componentId: "sc-1npw367-0"
})(["\n  display: flex;\n  flex-direction: column;\n  flex: 1 1 auto;\n  ", ";\n"], _flexMaxHeightIeFix.flexMaxHeightIEFix); // Header
// ==============================


exports.Wrapper = Wrapper;

var HeaderOrFooter = _styledComponents.default.div.withConfig({
  displayName: "Content__HeaderOrFooter",
  componentId: "sc-1npw367-1"
})(["\n  align-items: center;\n  display: flex;\n  flex: 0 0 auto;\n  justify-content: space-between;\n  transition: box-shadow 200ms;\n  z-index: 1;\n"]);

var Header = (0, _styledComponents.default)(HeaderOrFooter).withConfig({
  displayName: "Content__Header",
  componentId: "sc-1npw367-2"
})(["\n  padding: ", "px ", "px ", "px;\n  box-shadow: ", ";\n"], outerGutter, outerGutter, innerGutter - keylineHeight, function (p) {
  return p.showKeyline ? " 0 ".concat(keylineHeight, "px 0 0 ").concat(keylineColor(p)) : 'none';
}); // Title

exports.Header = Header;
var oneLineTitleText = (0, _styledComponents.css)(["\n  white-space: nowrap;\n  overflow: hidden;\n  text-overflow: ellipsis;\n"]);

var Title = _styledComponents.default.h4.withConfig({
  displayName: "Content__Title",
  componentId: "sc-1npw367-3"
})(["\n  align-items: center;\n  display: flex;\n  font-size: 20px;\n  font-style: inherit;\n  font-weight: 500;\n  letter-spacing: -0.008em;\n  line-height: 1;\n  margin: 0;\n  min-width: 0;\n"]);

exports.Title = Title;

var TitleText = _styledComponents.default.span.withConfig({
  displayName: "Content__TitleText",
  componentId: "sc-1npw367-4"
})(["\n  flex: 1 1 auto;\n  min-width: 0;\n  word-wrap: break-word;\n  width: 100%;\n  ", ";\n"], function (_ref) {
  var isHeadingMultiline = _ref.isHeadingMultiline;
  return !isHeadingMultiline && oneLineTitleText;
});

exports.TitleText = TitleText;
var iconColor = {
  danger: _theme.colors.R400,
  warning: _theme.colors.Y400
};

var TitleIconWrapper = _styledComponents.default.span.withConfig({
  displayName: "Content__TitleIconWrapper",
  componentId: "sc-1npw367-5"
})(["\n  color: ", ";\n  margin-right: ", "px;\n  flex: 0 0 auto;\n"], function (p) {
  return iconColor[p.appearance];
}, _theme.gridSize); // Body
// ==============================

/**
  Adding the padding here avoids cropping box shadow on first/last
  children. The combined vertical spacing is maintained by subtracting the
  keyline height from header and footer.
*/


exports.TitleIconWrapper = TitleIconWrapper;

var Body = _styledComponents.default.div.withConfig({
  displayName: "Content__Body",
  componentId: "sc-1npw367-6"
})(["\n  flex: 1 1 auto;\n  ", ";\n"], function (p) {
  return p.shouldScroll ? "\n          overflow-y: auto;\n          overflow-x: hidden;\n          padding: ".concat(keylineHeight, "px ").concat(outerGutter, "px;\n        ") : "\n          padding: 0 ".concat(outerGutter, "px;\n        ");
}); // Footer
// ==============================


exports.Body = Body;
var Footer = (0, _styledComponents.default)(HeaderOrFooter).withConfig({
  displayName: "Content__Footer",
  componentId: "sc-1npw367-7"
})(["\n  padding: ", "px ", "px ", "px;\n  box-shadow: ", ";\n"], innerGutter - keylineHeight, outerGutter, outerGutter, function (p) {
  return p.showKeyline ? "0 -".concat(keylineHeight, "px 0 0 ").concat(keylineColor(p)) : 'none';
});
exports.Footer = Footer;

var Actions = _styledComponents.default.div.withConfig({
  displayName: "Content__Actions",
  componentId: "sc-1npw367-8"
})(["\n  display: inline-flex;\n  margin: 0 -", "px;\n"], _theme.math.divide(_theme.gridSize, 2));

exports.Actions = Actions;

var ActionItem = _styledComponents.default.div.withConfig({
  displayName: "Content__ActionItem",
  componentId: "sc-1npw367-9"
})(["\n  flex: 1 0 auto;\n  margin: 0 ", "px;\n"], _theme.math.divide(_theme.gridSize, 2));

exports.ActionItem = ActionItem;
