// @atlaskit/modal-dialog/dist/cjs/styled/Modal.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Dialog = exports.PositionerRelative = exports.PositionerAbsolute = exports.FillScreen = exports.dialogHeight = exports.dialogWidth = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _theme = require("@atlaskit/theme");

var _sharedVariables = require("../shared-variables");

var _flexMaxHeightIeFix = require("../utils/flex-max-height-ie-fix");

var boxShadow = function boxShadow(_ref) {
  var isChromeless = _ref.isChromeless;
  return isChromeless ? 'none' : "\n    0 0 0 1px ".concat(_theme.colors.N30A, ", 0 2px 1px ").concat(_theme.colors.N30A, ",\n    0 0 20px -6px ").concat(_theme.colors.N60A, "\n  ");
};

var dialogBgColor = function dialogBgColor(_ref2) {
  var isChromeless = _ref2.isChromeless;
  return isChromeless ? 'transparent' : (0, _theme.themed)({
    light: _theme.colors.N0,
    dark: _theme.colors.DN50
  });
};

var maxDimensions = "calc(100% - ".concat(_sharedVariables.gutter * 2, "px)");
var maxHeightDimensions = "calc(100% - ".concat(_sharedVariables.gutter * 2 - _flexMaxHeightIeFix.IEMaxHeightCalcPx, "px)");

var dialogWidth = function dialogWidth(_ref3) {
  var widthName = _ref3.widthName,
      widthValue = _ref3.widthValue;

  if (typeof widthValue === 'number') {
    return "".concat(widthValue, "px");
  }

  return widthName ? "".concat(_sharedVariables.WIDTH_ENUM.widths[widthName], "px") : widthValue || 'auto';
};

exports.dialogWidth = dialogWidth;

var dialogHeight = function dialogHeight(_ref4) {
  var heightValue = _ref4.heightValue;

  if (typeof heightValue === 'number') {
    return "".concat(heightValue, "px");
  }

  return heightValue || 'auto';
};
/**
  NOTE:
  z-index
  - temporarily added to beat @atlaskit/navigation

  absolute + top
  - rather than fixed position so popper.js children are properly positioned

  overflow-y
  - only active when popper.js children envoked below the dialog
*/


exports.dialogHeight = dialogHeight;

var FillScreen = _styledComponents.default.div.withConfig({
  displayName: "Modal__FillScreen",
  componentId: "sc-1jmnqyd-0"
})(["\n  height: 100vh;\n  left: 0;\n  overflow-y: auto;\n  position: absolute;\n  top: ", "px;\n  width: 100%;\n  z-index: ", ";\n  -webkit-overflow-scrolling: touch;\n"], function (p) {
  return p.scrollDistance;
}, _theme.layers.modal);

exports.FillScreen = FillScreen;

var PositionerAbsolute = _styledComponents.default.div.withConfig({
  displayName: "Modal__PositionerAbsolute",
  componentId: "sc-1jmnqyd-1"
})(["\n  display: flex;\n  flex-direction: column;\n  height: ", ";\n  left: 0;\n  margin-left: auto;\n  margin-right: auto;\n  max-width: ", ";\n  position: absolute;\n  right: 0;\n  top: ", "px;\n  width: ", ";\n  z-index: ", ";\n  pointer-events: none;\n"], maxHeightDimensions, maxDimensions, _sharedVariables.gutter, dialogWidth, _theme.layers.modal);

exports.PositionerAbsolute = PositionerAbsolute;

var PositionerRelative = _styledComponents.default.div.withConfig({
  displayName: "Modal__PositionerRelative",
  componentId: "sc-1jmnqyd-2"
})(["\n  margin: ", "px auto;\n  position: relative;\n  width: ", ";\n  z-index: ", ";\n  pointer-events: none;\n"], _sharedVariables.gutter, dialogWidth, _theme.layers.modal);

exports.PositionerRelative = PositionerRelative;

var Dialog = _styledComponents.default.div.withConfig({
  displayName: "Modal__Dialog",
  componentId: "sc-1jmnqyd-3"
})(["\n  ", " color: ", ";\n  display: flex;\n  flex-direction: column;\n  height: ", ";\n  ", ";\n  outline: 0;\n  pointer-events: auto;\n"], function (p) {
  return p.isChromeless ? null : (0, _styledComponents.css)(["\n          background-color: ", ";\n          border-radius: ", "px;\n          box-shadow: ", ";\n        "], dialogBgColor, _theme.borderRadius, boxShadow);
}, _theme.colors.text, dialogHeight, _flexMaxHeightIeFix.flexMaxHeightIEFix);

exports.Dialog = Dialog;
PositionerAbsolute.displayName = 'PositionerAbsolute';
Dialog.displayName = 'Dialog';
FillScreen.displayName = 'FillScreen';
PositionerRelative.displayName = 'PositionerRelative';
