// @atlaskit/avatar/dist/cjs/components/AvatarImage.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DefaultImage = DefaultImage;
exports.default = exports.clearCache = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _theme = _interopRequireDefault(require("@atlaskit/theme"));

var _react = _interopRequireWildcard(require("react"));

var _AvatarImage = require("../styled/AvatarImage");

function DefaultImage(_ref) {
  var appearance = _ref.appearance,
      size = _ref.size,
      title = _ref.title,
      isLoading = _ref.isLoading;
  var rectBounds = 128;
  return _react.default.createElement(_theme.default.Consumer, null, function (_ref2) {
    var mode = _ref2.mode;
    return _react.default.createElement(_AvatarImage.Svg, {
      appearance: appearance,
      isLoading: isLoading,
      size: size,
      viewBox: "0 0 ".concat(rectBounds, " ").concat(rectBounds),
      version: "1.1",
      xmlns: "http://www.w3.org/2000/svg",
      role: "img",
      "aria-label": title
    }, appearance === 'circle' ? _react.default.createElement(_AvatarImage.ShapeGroup, {
      mode: mode
    }, _react.default.createElement("circle", {
      cx: "64",
      cy: "64",
      r: "64"
    }), _react.default.createElement("g", null, _react.default.createElement("path", {
      d: "M103,102.1388 C93.094,111.92 79.3504,118 64.1638,118 C48.8056,118 34.9294,111.768 25,101.7892 L25,95.2 C25,86.8096 31.981,80 40.6,80 L87.4,80 C96.019,80 103,86.8096 103,95.2 L103,102.1388 Z"
    }), _react.default.createElement("path", {
      d: "M63.9961647,24 C51.2938136,24 41,34.2938136 41,46.9961647 C41,59.7061864 51.2938136,70 63.9961647,70 C76.6985159,70 87,59.7061864 87,46.9961647 C87,34.2938136 76.6985159,24 63.9961647,24"
    }))) : _react.default.createElement(_AvatarImage.ShapeGroup, {
      mode: mode
    }, _react.default.createElement("rect", {
      x: "0",
      y: "0",
      width: rectBounds,
      height: rectBounds
    }), _react.default.createElement("g", {
      transform: "translate(19.000000, 32.000000)"
    }, _react.default.createElement("path", {
      d: "M18.25,32.5 L54.5833333,32.5 L54.5833333,23.4166667 L18.25,23.4166667 L18.25,32.5 Z M9.16666667,18.8331166 C9.16666667,16.3479549 11.236581,14.3333333 13.7195662,14.3333333 L59.1137671,14.3333333 C61.6282641,14.3333333 63.6666667,16.3815123 63.6666667,18.8331166 L63.6666667,41.5833333 L9.16666667,41.5833333 L9.16666667,18.8331166 Z"
    }), _react.default.createElement("path", {
      d: "M18.25,9.81383682 C18.25,4.7850061 22.3296003,0.708333333 27.3238554,0.708333333 L36.4261446,0.708333333 C41.4374965,0.708333333 45.5,4.76812825 45.5,9.81383682 L45.5,23.4166667 L18.25,23.4166667 L18.25,9.81383682 Z M36.4166667,9.81383682 C36.4166667,9.79803315 36.4184748,9.79303784 36.4207515,9.79166667 L27.3238554,9.79166667 C27.3447224,9.79166667 27.3333333,9.80308059 27.3333333,9.81383682 L27.3333333,14.3333333 L36.4166667,14.3333333 L36.4166667,9.81383682 Z"
    }), _react.default.createElement("path", {
      d: "M11.4386532,55.2083333 L74.9953562,55.2083333 L79.5452242,41.5833333 L9.80449752,41.5833333 L11.4386532,55.2083333 Z M0.1048547,36.987541 C-0.192399775,34.5091405 1.5865717,32.5 4.09502839,32.5 L87.6264735,32.5 C90.1274401,32.5 91.5225656,34.393506 90.7231047,36.7875656 L82.9702846,60.004101 C82.1795402,62.3720582 79.5279445,64.2916667 76.9985338,64.2916667 L7.91963924,64.2916667 C5.41227673,64.2916667 3.14113571,62.3029555 2.84143097,59.8041257 L0.1048547,36.987541 Z"
    }))));
  });
}

var cache = {};

var clearCache = function clearCache() {
  cache = {};
};

exports.clearCache = clearCache;

var AvatarImage =
/*#__PURE__*/
function (_PureComponent) {
  (0, _inherits2.default)(AvatarImage, _PureComponent);

  function AvatarImage() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, AvatarImage);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(AvatarImage)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "state", {
      hasError: false,
      // if provided a scr - we need to load it
      isLoading: Boolean(_this.props.src)
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "isComponentMounted", void 0);
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "loadImage", function () {
      // nothing to load
      if (!_this.props.src) {
        return;
      }

      var img = new Image();
      img.onload = _this.handleLoadSuccess;
      img.onerror = _this.handleLoadError;
      img.src = _this.props.src;
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "handleLoad", function (hasError) {
      if (_this.isComponentMounted) {
        _this.setState({
          hasError: hasError,
          isLoading: false
        });
      }
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "handleLoadSuccess", function () {
      if (typeof _this.props.src === 'string') {
        cache[_this.props.src] = true;
      }

      _this.handleLoad(false);
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "handleLoadError", function () {
      return _this.handleLoad(true);
    });
    return _this;
  }

  (0, _createClass2.default)(AvatarImage, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.isComponentMounted = true;
      this.loadImage();
    } // handle case where `src` is modified after mount

  }, {
    key: "componentWillReceiveProps",
    value: function componentWillReceiveProps(nextProps) {
      if (nextProps.src && this.props.src !== nextProps.src) {
        this.setState({
          isLoading: true
        });
      }
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      if (this.props.src && this.props.src !== prevProps.src) {
        this.loadImage();
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      this.isComponentMounted = false;
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          alt = _this$props.alt,
          src = _this$props.src,
          appearance = _this$props.appearance,
          size = _this$props.size;
      var _this$state = this.state,
          hasError = _this$state.hasError,
          isLoading = _this$state.isLoading;
      var showDefault = !isLoading && (!src || hasError);
      var imageUrl = src && (!isLoading || cache[src]) ? src : null;
      return showDefault ? _react.default.createElement(DefaultImage, {
        appearance: appearance,
        size: size,
        title: alt,
        isLoading: isLoading
      }) : _react.default.createElement(_AvatarImage.Slot, {
        appearance: appearance,
        isLoading: isLoading,
        size: size,
        role: "img",
        label: alt,
        backgroundImage: imageUrl
      });
    }
  }]);
  return AvatarImage;
}(_react.PureComponent);

exports.default = AvatarImage;
