// @atlaskit/avatar/dist/cjs/components/AvatarItem.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _toConsumableArray2 = _interopRequireDefault(require("@babel/runtime/helpers/toConsumableArray"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _theme = _interopRequireDefault(require("@atlaskit/theme"));

var _react = _interopRequireWildcard(require("react"));

var _constants = require("./constants");

var _utils = require("../utils");

var _AvatarItem = require("../styled/AvatarItem");

var _helpers = require("../helpers");

var _hoc = require("../hoc");

var _item = require("../theme/item");

var AvatarItem =
/*#__PURE__*/
function (_Component) {
  (0, _inherits2.default)(AvatarItem, _Component);

  function AvatarItem() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, AvatarItem);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(AvatarItem)).call.apply(_getPrototypeOf2, [this].concat(args)));
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "node", void 0);
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "blur", function () {
      if (_this.node) _this.node.blur();
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "focus", function () {
      if (_this.node) _this.node.focus();
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "guardedClick", function (event) {
      var _this$props = _this.props,
          isDisabled = _this$props.isDisabled,
          onClick = _this$props.onClick;
      if (isDisabled || typeof onClick !== 'function') return;

      var item = _utils.omit.apply(void 0, [_this.props].concat((0, _toConsumableArray2.default)(_constants.propsOmittedFromClickData)));

      onClick({
        item: item,
        event: event
      });
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)((0, _assertThisInitialized2.default)(_this)), "setNode", function (ref) {
      _this.node = ref;
    });
    return _this;
  }

  (0, _createClass2.default)(AvatarItem, [{
    key: "render",
    value: function render() {
      var _this2 = this;

      var _this$props2 = this.props,
          avatar = _this$props2.avatar,
          enableTextTruncate = _this$props2.enableTextTruncate,
          primaryText = _this$props2.primaryText,
          secondaryText = _this$props2.secondaryText; // distill props from context, props, and state

      var enhancedProps = (0, _helpers.getProps)(this); // provide element type based on props

      var StyledComponent = (0, _helpers.getStyledAvatarItem)(this.props);
      return _react.default.createElement(_theme.default.Consumer, null, function (_ref) {
        var mode = _ref.mode;
        return _react.default.createElement(_item.ThemeItem.Provider, {
          value: _this2.props.theme
        }, _react.default.createElement(_item.ThemeItem.Consumer, null, function (tokens) {
          // maintain the illusion of a mask around presence/status
          var borderColor = (0, _AvatarItem.getBackgroundColor)((0, _objectSpread2.default)({}, _this2.props, tokens, {
            mode: mode
          }));
          return _react.default.createElement(StyledComponent, (0, _extends2.default)({
            innerRef: _this2.setNode
          }, enhancedProps, {
            onClick: _this2.guardedClick
          }), (0, _react.cloneElement)(avatar, {
            borderColor: borderColor
          }), _react.default.createElement(_AvatarItem.Content, {
            truncate: enableTextTruncate
          }, _react.default.createElement(_AvatarItem.PrimaryText, {
            truncate: enableTextTruncate
          }, primaryText), _react.default.createElement(_AvatarItem.SecondaryText, {
            truncate: enableTextTruncate
          }, secondaryText)));
        }));
      });
    }
  }]);
  return AvatarItem;
}(_react.Component);

(0, _defineProperty2.default)(AvatarItem, "defaultProps", {
  enableTextTruncate: true
});

var _default = (0, _hoc.withPseudoState)(AvatarItem);

exports.default = _default;
