// @atlaskit/avatar/dist/cjs/styled/AvatarImage.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Svg = exports.Slot = exports.ShapeGroup = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _theme = require("@atlaskit/theme");

var _theme2 = require("../theme");

var ShapeGroup = (0, _theme.withTheme)(_styledComponents.default.g.withConfig({
  displayName: "AvatarImage__ShapeGroup",
  componentId: "sc-1hf6c4m-0"
})(["\n  & circle,\n  & rect {\n    fill: ", ";\n  }\n  & g {\n    fill: ", ";\n  }\n"], (0, _theme.themed)({
  light: _theme.colors.N50,
  dark: _theme.colors.DN100
}), _theme.colors.background));
exports.ShapeGroup = ShapeGroup;

var Slot = function Slot(_ref) {
  var isLoading = _ref.isLoading,
      appearance = _ref.appearance,
      size = _ref.size,
      backgroundImage = _ref.backgroundImage,
      label = _ref.label,
      role = _ref.role;
  return _react.default.createElement(_theme2.Theme.Consumer, {
    appearance: appearance,
    isLoading: isLoading,
    size: size
  }, function (_ref2) {
    var backgroundColor = _ref2.backgroundColor,
        borderRadius = _ref2.borderRadius;
    return _react.default.createElement("span", {
      style: {
        backgroundColor: backgroundColor,
        backgroundImage: backgroundImage ? "url(".concat(backgroundImage, ")") : null,
        backgroundPosition: 'center',
        backgroundRepeat: 'no-repeat',
        backgroundSize: 'cover',
        borderRadius: borderRadius,
        display: 'flex',
        flex: '1 1 100%',
        height: '100%',
        width: '100%'
      },
      role: role,
      "aria-label": label
    });
  });
};

exports.Slot = Slot;

var Svg = function Svg(_ref3) {
  var appearance = _ref3.appearance,
      size = _ref3.size,
      children = _ref3.children,
      isLoading = _ref3.isLoading,
      otherProps = (0, _objectWithoutProperties2.default)(_ref3, ["appearance", "size", "children", "isLoading"]);
  return _react.default.createElement(_theme2.Theme.Consumer, {
    appearance: appearance,
    isLoading: isLoading,
    size: size
  }, function (_ref4) {
    var backgroundColor = _ref4.backgroundColor,
        borderRadius = _ref4.borderRadius;
    return _react.default.createElement("svg", (0, _extends2.default)({
      style: {
        backgroundColor: backgroundColor,
        borderRadius: borderRadius,
        height: '100%',
        width: '100%'
      }
    }, otherProps), children);
  });
};

exports.Svg = Svg;
