// @atlaskit/avatar/dist/cjs/styled/Avatar.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StatusWrapper = exports.PresenceWrapper = exports.Inner = exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _theme = require("@atlaskit/theme");

var _theme2 = require("../theme");

var _utils = require("./utils");

var _default = function _default(props) {
  return _react.default.createElement(_theme2.Theme.Consumer, (0, _extends2.default)({}, props, {
    includeBorderWidth: true
  }), function (_ref) {
    var dimensions = _ref.dimensions;
    return _react.default.createElement("div", {
      style: (0, _objectSpread2.default)({
        display: 'inline-block',
        position: 'relative',
        outline: 0,
        zIndex: props.stackIndex
      }, dimensions)
    }, props.children);
  });
}; // TODO this doesn't appear to be used anywhere so we should look at removing.


exports.default = _default;
var Inner = (0, _theme.withTheme)(_styledComponents.default.div.withConfig({
  displayName: "Avatar__Inner",
  componentId: "r9c9gb-0"
})(["\n  ", ";\n"], _utils.getInnerStyles));
exports.Inner = Inner;

var PresenceWrapper = function PresenceWrapper(props) {
  return _react.default.createElement(_theme2.Theme.Consumer, (0, _extends2.default)({}, props, {
    includeBorderWidth: true
  }), function (_ref2) {
    var presence = _ref2.presence;
    return _react.default.createElement("span", {
      style: (0, _objectSpread2.default)({
        pointerEvents: 'none',
        position: 'absolute'
      }, presence)
    }, props.children);
  });
};

exports.PresenceWrapper = PresenceWrapper;

var StatusWrapper = function StatusWrapper(props) {
  return _react.default.createElement(_theme2.Theme.Consumer, (0, _extends2.default)({}, props, {
    includeBorderWidth: true
  }), function (_ref3) {
    var status = _ref3.status;
    return _react.default.createElement("span", {
      style: (0, _objectSpread2.default)({
        position: 'absolute'
      }, status)
    }, props.children);
  });
};

exports.StatusWrapper = StatusWrapper;
