// @atlaskit/avatar/dist/cjs/styled/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SQUARE_ICON_OFFSET = exports.ICON_OFFSET = exports.ICON_SIZES = exports.EXCESS_INDICATOR_FONT_SIZE = exports.BORDER_WIDTH = exports.AVATAR_RADIUS = exports.AVATAR_SIZES = exports.TRANSITION_DURATION = void 0;

var _theme = require("@atlaskit/theme");

var TRANSITION_DURATION = '200ms';
exports.TRANSITION_DURATION = TRANSITION_DURATION;
var gridSizeValue = (0, _theme.gridSize)();
var AVATAR_SIZES = {
  xsmall: gridSizeValue * 2,
  small: gridSizeValue * 3,
  medium: gridSizeValue * 4,
  large: gridSizeValue * 5,
  xlarge: gridSizeValue * 12,
  xxlarge: gridSizeValue * 16
}; // border radius only applies to "square" avatars

exports.AVATAR_SIZES = AVATAR_SIZES;
var AVATAR_RADIUS = {
  xsmall: 2,
  small: 2,
  medium: 3,
  large: 3,
  xlarge: 6,
  xxlarge: 12
};
exports.AVATAR_RADIUS = AVATAR_RADIUS;
var BORDER_WIDTH = {
  xsmall: 2,
  small: 2,
  medium: 2,
  large: 2,
  xlarge: 2,
  xxlarge: 2
}; // NOTE: sizes xsmall & xxlarge DO NOT support
// - groups
// - presence
// - status

exports.BORDER_WIDTH = BORDER_WIDTH;
var EXCESS_INDICATOR_FONT_SIZE = {
  small: 10,
  medium: 11,
  large: 12,
  xlarge: 16
};
exports.EXCESS_INDICATOR_FONT_SIZE = EXCESS_INDICATOR_FONT_SIZE;
var ICON_SIZES = {
  small: 12,
  medium: 14,
  large: 15,
  xlarge: 18
};
exports.ICON_SIZES = ICON_SIZES;
var ICON_OFFSET = {
  small: 0,
  medium: 0,
  large: 1,
  xlarge: 7
};
exports.ICON_OFFSET = ICON_OFFSET;
var SQUARE_ICON_OFFSET = {
  small: 0,
  medium: 0,
  large: 0,
  xlarge: 1
};
exports.SQUARE_ICON_OFFSET = SQUARE_ICON_OFFSET;
