// @atlaskit/blanket/dist/cjs/styled.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.pointerEvents = exports.opacity = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _theme = require("@atlaskit/theme");

var backgroundColor = (0, _theme.themed)({
  light: _theme.colors.N100A,
  dark: _theme.colors.DN90A
});

var opacity = function opacity(p) {
  return p.isTinted ? 1 : 0;
};

exports.opacity = opacity;

var pointerEvents = function pointerEvents(p) {
  return p.canClickThrough ? 'none' : 'initial';
};

exports.pointerEvents = pointerEvents;

var _default = _styledComponents.default.div.withConfig({
  displayName: "styled",
  componentId: "knyj0u-0"
})(["\n  background: ", ";\n  bottom: 0;\n  left: 0;\n  opacity: ", ";\n  pointer-events: ", ";\n  position: fixed;\n  right: 0;\n  top: 0;\n  transition: opacity 220ms;\n  z-index: ", ";\n"], backgroundColor, opacity, pointerEvents, _theme.layers.blanket);

exports.default = _default;
